File: algorithms/1_easy/S/01221_split_a_string_in_balanced_strings/solution.cpp
class Solution {
public:
    int balancedStringSplit(string s) {

        int count{0};
        int curr_diff{0};

        for (const char &ch: s) {
            if (ch == 'R') {
                curr_diff += 1;
            } else {
                curr_diff -= 1;
            }

            if (curr_diff == 0) {
                count += 1;
            }
        }

        return count;
    }
};

File: algorithms/1_easy/S/01221_split_a_string_in_balanced_strings/solution.rs
impl Solution {
    pub fn balanced_string_split(s: String) -> i32 {

        let mut count = 0;
        let mut curr_diff = 0;

        for byte in s.into_bytes().into_iter() {
            if byte == b'R' {
                curr_diff += 1;
            } else {
                curr_diff -= 1;
            }

            if curr_diff == 0 {
                count += 1;
            }
        }

        count
    }
}
